src/storage/database_sink.rs

```diff
@@ -429,6 +431,16 @@ impl<'s> DatabaseSink<'s> {
         let unit_index_statement = self.connection.prepare(&unit_index_statement)?;
         self.connection.execute(&unit_index_statement, &[])?;
 
+        let geom_index_statement = format!(
+            "CREATE INDEX {unit_table}_geom_idx ON {schema}.{unit_table} \
+             USING GIST (geom);",
```

Comment:
Since our data has no overlaps in the spatial dimension (since we have points), we should use SPGIST I guess.
https://blog.crunchydata.com/blog/the-many-spatial-indexes-of-postgis

Reply:
dbeeec65b9484ae89be714f6a9106092ada0a7ec